lib/ui/common/roundedButton.dart

```diff
@@ -57,14 +58,14 @@ class RoundedButton {
       {RoundedButtonStyle style = const _DefaultStyle(),
         @required  BuildContext context,
         String title, IconData icon,
-        Function onTap}) {
-    return _buildRoundedButton(style, context , title: title, icon: icon, onTap: onTap);
+        Function onTap, Color backgroundColor}) {
+    return _buildRoundedButton(style, context , title: title, icon: icon, onTap: onTap, backgroundColor: backgroundColor);
   }
 
   static Widget _buildRoundedButton(RoundedButtonStyle buttonStyle, BuildContext context, {String title, IconData icon,
-    Function onTap}) {
+    Function onTap, Color backgroundColor}) {
 
-    RoundedButtonViewModel viewModel = buildButtonViewModel(context, title: title, icon: icon, onTap: onTap);
+    RoundedButtonViewModel viewModel = buildButtonViewModel(context, title: title, icon: icon, onTap: onTap, backgroundColor: backgroundColor);
```

Comment:
why not just pass in the viewModel to this function?

Reply:
Now button view model is passed to the build function